protocol/common/message_linkpreview.go

```diff
@@ -468,3 +470,26 @@ func (m *Message) ConvertFromProtoToStatusLinkPreviews(makeMediaServerURL func(m
 
 	return previews
 }
+
+func decompressPublicKey(input string) string {
+	decompressed, err := multiformat.DeserializePublicKey(input, "f")
+	if err != nil {
+		return input
+	}
+	const secp256k1Code = "fe701"
+	return "0x" + strings.TrimPrefix(decompressed, secp256k1Code)
+}
+
+func (m *Message) DecompressPublicKeys() {
+	for _, linkPreview := range m.StatusLinkPreviews {
+		if linkPreview.Contact != nil {
+			linkPreview.Contact.PublicKey = decompressPublicKey(linkPreview.Contact.PublicKey)
```

Comment:
In a way, this feels like a workaround no? The public key is already compressed because that's what we want clients to use. Maybe it's totally fine, I'd just confirm with @cammellos

Reply:
Ideally we move away from decompressed keys and we use compressed keys everywhere, we can make status-go endpoints agnostic, though it's still a lot of work to get rid of them completely, so maybe do it gradually is best, i guess it depends on why we need it, and we can make a decision?

Reply:
In status-desktop we have mostly work with decompressed keys, e.g. there're many maps like `map[decompressedKey]Community`. 

Also, we store decompressed keys in the database. So I thought that it's the main format we use.

@cammellos @ilmotta 
I'm fine with changing it everywhere, but that sounds like a different issue to do. 
Do you think this this PR makes sense for now?

> though it's still a lot of work to get rid of them completely

And this very true 😄

Reply:
hard to say, I don't want you to commit to a bunch of extra work that is not strictly required, and similarly on mobile we do exactly the same, we index by uncompressed key. I am not sure it would be a priority to make the swap, and I can imagine being quite painful, so I would leave that decision up to you, you'd know best the complexity of the change against the priorities.

Reply:
I'm very sure making this change is much simpler for now 😄